File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.js
/**
 * @param {number} c
 * @return {boolean}
 */
var judgeSquareSum = function(c) {
    
    let i =0 , j =Math.floor( Math.sqrt(c) ) ;
    while (i<= j)
    {
        if( i*i + j*j == c )
            return true;
        else if(i*i + j*j > c )
            j--;
        else if (i*i + j*j < c )
            i++;
    }
    return false;
    
};

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) 
{
    int i =0 , j = sqrt(c) ;
    while (i<= j)
    {
        if( 1LL*i*i + 1LL*j*j == c )
            return true;
        else if(1LL*i*i + 1LL*j*j > c )
            j--;
        else if (1LL*i*i + 1LL*j*j < c )
            i++;
    }
    return false;
}
};